### DISCUSSION TITLE
Concrete link between image and the configuration that produced it

### DISCUSSION THREAD

**Discussion by joshuagl** [actions/virtual-environments#5483] [Ideas]
_Note_: this is a feature request but does not match the [current issue template](https://github.com/actions/virtual-environments/issues/new?assignees=&labels=feature+request%2Cneeds+triage&template=tool-request.yml&title=Update%2FAdd+%5Btool+name%5D).

__Feature request__: can we include the commit digest of the release tag the image was built from in the image? (i.e. environment variable or [/imagegeneration/imagedata.json](https://github.com/actions/virtual-environments/blob/b1fae5fe509257684309c195c0a1d713e830e025/images/linux/ubuntu2004.json#L18)).

Context: I work on the [SLSA project] and [slsa-github-generator-go], where we are building tools to support producing SLSA level 3 artefacts with GitHub Actions. (See GitHub blog post "[Achieving SLSA 3 Compliance with GitHub Actions and Sigstore for Go modules]" and Google blog post "[Improving software supply chain security with tamper-proof builds]" for more information).

We include a link to the virtual-environments release in the [`materials`] section of the SLSA provenance we generate and would also like to include a corresponding digest for that material. However, AFAICT the virtual-environment images do not include the SHA of the commit the release tag references.

We _could_ resolve the tag to a commit SHA at provenance generation time (i.e. `gh api -H "Accept: application/vnd.github.v3+json" /repos/actions/virtual-environments/git/matching-refs/tags/$ImageOS/$ImageVersion`), but as tags can be overwritten – and therefore may not resolve to the same commit digest at API call time – this doesn't feel appropriate.

I'd be more than willing to submit a PR for this if the feature is agreeable, implementation pointers would be gratefully received.

[SLSA project]: https://slsa.dev
[slsa-github-generator-go]: https://github.com/slsa-framework/slsa-github-generator-go
[Achieving SLSA 3 Compliance with GitHub Actions and Sigstore for Go modules]: https://github.blog/2022-04-07-slsa-3-compliance-with-github-actions/
[Improving software supply chain security with tamper-proof builds]: https://security.googleblog.com/2022/04/improving-software-supply-chain.html
[`materials`]: https://slsa.dev/provenance/v0.2#materials

**Comment by miketimofeev** [comment]
Hi @joshuagl!
The problem is that the commit might be misleading as lots of software is installed using the latest available version at the moment of the image generation. That is the reason why we don't recommend sticking to any releases and use always the main branch.

**Comment by joshuagl** [comment]
Hello @miketimofeev !

Is it possible, and indeed deemed worthwhile, to augment the images building to capture more version information about which versions are installed at image creation? (certainly yes for Linux images, but I'm less certain about non-Linux OS).

Per the SLSA model it is useful to have full information about the materials which are used in a build for policy and remediation (plus reproducibility, but that is a) not needed until SLSA level 4 and b) may not be possible for the images here given fetching latest at image generation).

**Reply by miketimofeev** [reply]
Well, our readme files are quite detailed and we try to include all the software versions in them, but any improvements are welcome
